src/webapp/logic/sheetImport.js

```diff
@@ -5,6 +5,67 @@ import i18n from "../../locales";
 import { stringEquals } from "../utils/strings";
 import { getObjectVersion } from "./utils";
 
+// Algorithm based on:
+// https://wrf.ecse.rpi.edu//Research/Short_Notes/pnpoly.html
+function checkCoordinates(coord, vs) {
+    const x = coord.longitude,
+        y = coord.latitude;
+
+    for (let l = 0; l < vs.length; ++l) {
+        for (let k = 0; k < vs[l].length; ++k) {
+            let inside = false;
+            for (let i = 0, j = vs[l][k].length - 1; i < vs[l][k].length; j = i++) {
+                const xi = vs[l][k][i][0],
+                    yi = vs[l][k][i][1];
+                const xj = vs[l][k][j][0],
+                    yj = vs[l][k][j][1];
+
+                const intersect = yi > y !== yj > y && x < ((xj - xi) * (y - yi)) / (yj - yi) + xi;
+                if (intersect) inside = !inside;
+            }
+            if (inside) return true;
+        }
+    }
+
+    return false;
+}
```

Comment:
This should be probably moved into a repository, code complexity requires at least some tests to ensure it works as expected.